pol-core/bscript/compiler/codegen/CodeGenerator.cpp

```diff
@@ -118,9 +114,20 @@ void CodeGenerator::generate_instructions( CompilerWorkspace& workspace )
     }
   }
 
+  // The default parameter generation needs to happen after generating the
+  // instructions for the user functions, as only class method functions and
+  // user functions with a registered function reference emit their default
+  // arguments. Since a user function may be registered as a function reference
+  // only _during_ some other user function's generation, we are only sure a
+  // user function has a function reference after generating all of them.
+  for ( const auto& user_function : workspace.user_functions )
```

Comment:
This adds another `O(n)` loop at the benefit of having smaller ecl program size. 

In this implementation, a script with 100 user functions with only one as a function reference will emit default argument instructions for that _one_ function (but still loops through all 100 functions to check if it's referenced).

We can remove this second-pass loop by emitting the default arguments for _all_ user functions, but then we could have a gigantic ECL, as all user functions -- regardless of `@FuncRef` usage -- would have their default arguments emitted.